tree_test.go

```diff
@@ -1806,3 +1806,36 @@ func TestRandom(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestRandomExtracted(t *testing.T) {
```

Comment:
So after a while of debugging `TestRandom(...)` I extracted (and manually "minified") a simple test that surfaced the bug. Leaving the test here to avoid regressions.

Interestingly, this bug wasn't affecting [this other test](https://github.com/ethereum/go-verkle/blob/master/tree_test.go#L290-L358) that was targeting this API.